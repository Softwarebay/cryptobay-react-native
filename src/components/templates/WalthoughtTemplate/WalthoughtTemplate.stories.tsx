import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { WalthoughtTemplate } from './WalthoughtTemplate';

storiesOf('Templates/WalthoughtCarousel', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Walthought Template', () => (
    <WalthoughtTemplate onNavigate={action('onNavigate')} />
  ));
