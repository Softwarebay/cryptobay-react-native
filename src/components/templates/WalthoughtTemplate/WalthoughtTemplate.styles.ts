import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  walthoughtTemplate: {
    flex: 1,
    width: '100%',
    backgroundColor: '#000000',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  contentWrapper: {
    padding: 30,
    width: '100%',
  },
});
