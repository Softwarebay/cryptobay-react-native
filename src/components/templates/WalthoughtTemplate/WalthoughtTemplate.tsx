import React from 'react';
import { SafeAreaView, View } from 'react-native';

import { WalthoughtCarousel } from '../../organisms';

import { styles } from './WalthoughtTemplate.styles';

type WalthoughtTemplateProps = {
  onNavigate: () => void;
};

export const WalthoughtTemplate: React.FC<WalthoughtTemplateProps> = ({
  onNavigate,
}: WalthoughtTemplateProps) => {
  return (
    <SafeAreaView style={styles.walthoughtTemplate}>
      <View style={styles.contentWrapper}>
        <WalthoughtCarousel onNavigate={onNavigate} />
      </View>
    </SafeAreaView>
  );
};
