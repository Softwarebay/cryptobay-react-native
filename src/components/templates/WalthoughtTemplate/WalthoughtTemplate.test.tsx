import React from 'react';
import '@testing-library/jest-native';
import 'react-native';
import renderer from 'react-test-renderer';

import { WalthoughtTemplate } from './WalthoughtTemplate';

const mockOnCarouselSlidesEnd = jest.fn();

describe('WalthoughtTemplate Template', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <WalthoughtTemplate onCarouselSlidesEnd={mockOnCarouselSlidesEnd} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
