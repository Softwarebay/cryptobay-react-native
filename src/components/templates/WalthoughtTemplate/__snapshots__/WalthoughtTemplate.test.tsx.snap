// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalthoughtTemplate Template Component Snapshot 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#17171A",
      "flex": 1,
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "padding": 30,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/pictures/cone.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 302,
          }
        }
        testID="walthought-carousel-picture"
      />
      <RNCMaskedView>
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "lineHeight": 56,
                },
                Object {
                  "fontSize": 40,
                  "fontWeight": "400",
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginBottom": 62,
                  "marginTop": 32,
                },
              ]
            }
            testID="walthought-carousel-header"
          >
            Property Diversity
          </Text>
        </View>
        <BVLinearGradient
          colors={
            Array [
              4289318399,
              4285724369,
              4288736653,
              4294890341,
              4294616222,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 0,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                  "lineHeight": 56,
                },
                Object {
                  "fontSize": 40,
                  "fontWeight": "400",
                },
                Object {
                  "opacity": 0,
                },
                Object {
                  "alignSelf": "flex-start",
                  "marginBottom": 62,
                  "marginTop": 32,
                },
              ]
            }
          >
            Property Diversity
          </Text>
        </BVLinearGradient>
      </RNCMaskedView>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "marginBottom": 16,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 8,
                "height": 8,
                "marginHorizontal": 4,
                "width": 8,
              },
              Object {
                "backgroundColor": "#FEBF32",
              },
            ]
          }
          testID="walthought-carousel-slide-1-dot"
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 8,
                "height": 8,
                "marginHorizontal": 4,
                "width": 8,
              },
              Object {
                "backgroundColor": "#4C516B",
              },
            ]
          }
          testID="walthought-carousel-slide-2-dot"
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 8,
                "height": 8,
                "marginHorizontal": 4,
                "width": 8,
              },
              Object {
                "backgroundColor": "#4C516B",
              },
            ]
          }
          testID="walthought-carousel-slide-3-dot"
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2A2D3C",
            "borderRadius": 8,
            "color": "#FEBF32",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 16,
            "width": "100%",
          }
        }
        testID="walthought-carousel-button"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
              },
              Object {
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 24,
              },
              Object {
                "color": "#FEBF32",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
